contracts/contracts/eosio.system/src/voting.cpp

```diff
@@ -18,26 +18,31 @@
 #include <cmath>
 
 namespace eosiosystem {
-   
    using eosio::const_mem_fun;
    using eosio::indexed_by;
    using eosio::singleton;
    using eosio::transaction;
+   using namespace std::literals::string_literals;
+
+
 
    void system_contract::regproducer( const name& producer, const eosio::public_key& producer_key, const std::string& url, uint16_t location ) {
+      require_auth( producer );
       check( url.size() < 512, "url too long" );
       check( producer_key != eosio::public_key(), "public key should not be the default value" );
-      require_auth( producer );
+
+      // TODO implement as constexpr string
+      static const auto stake_err = "user should stake at least "s + std::to_string( producer_stake_threshold ) + " REM to become a producer"s;
```

Comment:
per Roman request producer_stake_threshold should not be constant, please remove static prefix as it will lead to really hard to catch issue